plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/LocalizationPlugin.java

```diff
@@ -0,0 +1,114 @@
+package com.mapbox.mapboxsdk.plugins.testapp.activity;
+
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * The localization plugin enables localization of map labels into the user’s preferred language
+ * <p>
+ * Initialise this plugin in the {@link com.mapbox.mapboxsdk.maps.OnMapReadyCallback#onMapReady(MapboxMap)} and provide
+ * a valid instance of a {@link MapboxMap}.
+ * </p>
+ */
+public final class LocalizationPlugin {
+
+  private List<Layer> listOfMapLayers;
+
+  /**
+   * Create a localization plugin.
+   *
+   * @param mapboxMap the MapboxMap to apply localization plugin with
+   * @since 0.1.0
+   */
+  public LocalizationPlugin(@NonNull final MapboxMap mapboxMap) {
+    listOfMapLayers = mapboxMap.getLayers();
+    String deviceLanguage = Locale.getDefault().getLanguage();
+    for (Layer layer : listOfMapLayers) {
```

Comment:
What if the style includes layers that use a source other than Mapbox Streets? For example, there may be a layer based on a custom Studio dataset that also has a `{name}` property but no corresponding `{name_*}` properties.

Reply:
Now checking for Mapbox-only sources with this commit

https://github.com/mapbox/mapbox-plugins-android/pull/74/commits/0314d08e6d91a87f66aae1adab1981deed6cc740

Reply:
It turns out that commit doesn’t actually filter by source. Instead, it reruns the code once for each source, meaning that each layer gets processed one time per source. The Android map SDK apparently lacks a way to determine the source that a layer is using: mapbox/mapbox-gl-native#12691.